tgui/packages/tgui/interfaces/Wizard_Spellbook.js

```diff
@@ -0,0 +1,171 @@
+  const [searchQuery, setSearchQuery] = useLocalState(context, 'searchQuery', '');
+  const {
+    spellbook_contents,
+    spell_slots,
+    owner_name,
+  } = data;
+
+  let spell_categories = [];
+  for (let spell_category in spellbook_contents) {
+    spell_categories.push(spell_category);
+  }
+
+  return (
+    <Window
+      title={"Wizard Spellbook"}
+      theme={"ntos"}
+      fontSize={2}
+      height={600}
+      width={500}
+    >
+      <Window.Content>
+        <Section title={owner_name+"'s Spellbook "}>
+          <Flex justify={"space-between"}>
+            <Flex.Item>
+              {"Spell slots remaining: "+spell_slots}
+            </Flex.Item>
+            <Flex.Item>
+              <Input
+                value={searchQuery}
+                placeholder={"search by spell name"}
+                width={15}
+                autoSelect
+                onInput={(_, value) => setSearchQuery(value)}
+              />
+            </Flex.Item>
+          </Flex>
+        </Section>
+      </Window.Content>
+      <Window.Content mt={12} scrollable>
+        <Section>
+          {spell_categories.map((category) => (
+            <SpellCategory category={category} searchQuery={searchQuery} key={category} />
+          ))}
+        </Section>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const SpellCategory = (props, context) => {
+  const { data } = useBackend(context);
+  const { spellbook_contents } = data;
+  const { category, searchQuery } = props;
+
+  let spells = [];
+  for (let spell_name in spellbook_contents[category]) {
+    spells.push(spell_name);
+  }
+
+  return (
+    <Collapsible
+      title={category}
+      textColor={category_coloring(category)}
+      bold
+    >
+      <Stack vertical>
+        {spells.filter((spell) => spell.toLowerCase().includes(searchQuery.toLowerCase())).map((spell) => (
+          <Spell spell={spell} category={category} key={spell} />
+        ))}
+      </Stack>
+    </Collapsible>
+  );
+};
+
+const If_Purchased_Text = (purchased, cost, spell_slots) => {
+  if (purchased) {
+    return "Spell purchased";
+  } else if (cost > spell_slots) {
+    return "Not enough spell slots";
+  } else {
+    return "Purchase";
+  }
+};
+
+const Spell = (props, context) => {
+  const { data, act } = useBackend(context);
+  const {
+    spellbook_contents,
+    spell_slots,
+    vr,
+  } = data;
+  const { spell, category } = props;
+
+  const [purchased, setPurchased] = useSharedState(context, spell, false);
+  let spell_contents = []; // Non-associated list of: desc, cost, cooldown, vr_allowed
+  for (let spell_data in spellbook_contents[category][spell]) {
+    spell_contents.push(spellbook_contents[category][spell][spell_data]);
+  }
+
+  return (
+    <Stack.Item>
+      <Section>
+        {(vr === 1) && (spell_contents[3] === 0) && (
+          <Dimmer>
+            <Box fontSize={1.5} backgroundColor={"#384e68"} p={2}>
+              spell unavailable in VR
```

Comment:
```suggestion
              Spell unavailable in VR
```
capitalization might look nicer